brave-grpc/src/main/java/com/github/kristofa/brave/grpc/BraveGrpcServerInterceptor.java

```diff
@@ -90,7 +93,14 @@ public String getSpanName() {
 
         @Override
         public Collection<KeyValueAnnotation> requestAnnotations() {
-            return Collections.emptyList();
+            SocketAddress socketAddress = call.attributes().get(ServerCall.REMOTE_ADDR_KEY);
+            if (socketAddress != null) {
+                KeyValueAnnotation remoteAddrAnnotation = KeyValueAnnotation.create(
```

Comment:
The [ServerRequestInterceptor](https://github.com/openzipkin/brave/blob/master/brave-core/src/main/java/com/github/kristofa/brave/ServerRequestInterceptor.java#L49) makes the call to `setServerReceived()`.  Because of the way the [ServerRequestAdapter](https://github.com/openzipkin/brave/blob/master/brave-core/src/main/java/com/github/kristofa/brave/ServerRequestAdapter.java) interface is setup, there is not a hook to set the client address annotation prior to `setServerReceived()` being called.

If you think it makes sense, I can add an additional method to `ServerRequestAdapter` that allows implementors to optionally supply the client info so it can be used by `ServerRequestInterceptor`.  We could also use this to enhance the `HttpServerRequestAdapter` as well using the various methods to pull client IPs ([ServletRequest.html#getRemoteAddr()](i.e. http://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getRemoteAddr())and well known proxy headers such as `X-Forwarded-For`)